src/main/scala/org/apache/mesos/chronos/scheduler/jobs/Schedule.scala

```diff
@@ -1,44 +1,162 @@
+/**
+ * A Unix Cron schedule
+ *
+ * @param schedule the text representation of this schedule
+ * @param scheduleTimeZone the time zone the schedule should be executed from
+ * @param invocationTime the time to execute this schedule (UTC)
+ * @param lastExecutionTime the last time this schedule was run (UTC)
+ * @param cron the parsed Cron object that is used to perform execution time planning
+ */
+case class CronSchedule(@JsonProperty schedule: String,
+                        @JsonProperty scheduleTimeZone: String,
+                        @JsonProperty invocationTime: DateTime,
+                        @JsonProperty lastExecutionTime: DateTime,
+                         cron: Cron) extends Schedule {
+  type SType = CronSchedule
+  val recurrences = None
+  val scheduleType = CronType
+
+  lazy val next: Option[CronSchedule] = {
+    val tz = if(scheduleTimeZone != "") DateTimeZone.forID(scheduleTimeZone) else DateTimeZone.UTC
+    val currentTimeUserTZ = invocationTime.withZone(tz)
+    val nextExecUserTZ = ExecutionTime.forCron(cron).nextExecution(currentTimeUserTZ)
+    val nextExec = nextExecUserTZ.withZone(DateTimeZone.UTC)
+
+    Some(copy(invocationTime = nextExec, lastExecutionTime = invocationTime))
+  }
+
+  def toStringRepresentation = cron.asString()
+}
+
+object Schedules {
+  private val log = Logger.getLogger(getClass.getName)
+  private val cronDefinition = CronDefinitionBuilder.instanceDefinitionFor(CronTypeEnum.UNIX)
+
+  /**
+   * Parses a text representation of a schedule into a Schedule object, or None if the parsing fails
+   * @param scheduleStr the string representation of the schedule
+   * @param timeZoneStr the time zone to execute the schedule in
+   * @param scheduleType the type of schedule (Iso8601Type or CronType)
+   * @param currentTime the current DateTime in UTC
+   * @return
+   */
+  def parse(scheduleStr: String,
+            timeZoneStr: String = "",
+            scheduleType: ScheduleType = Iso8601Type,
+            currentTime: DateTime = DateTime.now(DateTimeZone.UTC)): Option[Schedule] = {
+
+    scheduleType match {
+      case Iso8601Type => parseIso8601Schedule(scheduleStr, timeZoneStr)
+      case CronType => parseCronSchedule(scheduleStr, timeZoneStr, currentTime)
+    }
+  }
+
+  def parseIso8601Schedule(scheduleStr: String, timeZoneStr: String = ""): Option[Iso8601Schedule] = {
     Iso8601Expressions.parse(scheduleStr, timeZoneStr).map {
       case (recurrences, start, period) =>
-        Schedule(scheduleStr, timeZoneStr, start, start, offset = 0, if (recurrences == -1) None else Some(recurrences), period)
+        Iso8601Schedule(scheduleStr, timeZoneStr, start, start, offset = 0, if (recurrences == -1) None else Some(recurrences), period)
     }
   }
 
+  def parseCronSchedule(scheduleStr: String, timeZoneStr: String = "UTC", currentTime: DateTime = DateTime.now(DateTimeZone.UTC)): Option[CronSchedule] = {
+    Try(parseCron(scheduleStr)) match {
```

Comment:
Why not use a more traditional `try` ... `catch` instead of a pattern match on `Try`?